### DISCUSSION TITLE
How can I set a global option for an extension used in multiple quarto documents?

### DISCUSSION THREAD

**Discussion by coatless** [quarto-dev/quarto-cli#6968] [Q&A] [answered]
### Description

Suppose I have a collection of analysis reports, each using the same extension with identical YAML options. For instance, the `highlight` extension facilitates text region highlighting by specifying a translucent background color, as seen in the sample YAML below:

```md
----
highlight:
  background-color: "blue"
  transparency: 25
filters:
  - highlight
----
```

Is there a way to set a global extension option within `_quarto.yml` or a dedicated extension file at the project's top level that the Lua filter could access? Additionally, is there an effective approach for defining extension-specific parameters that are allowed within the extension's metadata key, without relying on runtime checks to catch invalid keys during rendering?

**Comment by dragonstyle** [comment, marked as answer]
- For the first question, placing the `highlight` key and subkeys should work within a `_quarto.yml` file or within a `_metadata.yml` file. Those values will be merged into the document metadata and available to filters.

- For the second question, there is currently isn't a way to perform validation without runtime checks that I'm aware.

**Reply by cscheid** [reply]
> For the second question, there is currently isn't a way to perform validation without runtime checks that I'm aware.

That's right. We (internally) discussed the possibility of exposing YAML validation for extension-specific schemas, but there's a number of legitimate complications on how to make this work on IDEs. We'll have the feature eventually but right now there's no great way to do it.

**Reply by coatless** [reply]
Oh wow, that's nice! I didn't quite realize extensions were also supported there. I thought that was locked down due to the schema validation already taking place on `_quarto.yml`. I'll probably submit a PR to add that as a use case in the [Project Documentation](https://quarto.org/docs/projects/quarto-projects.html). 

It's also great to hear that schema validation will slowly make its way over to extensions.

**Reply by dragonstyle** [reply]
We'd appreciate the PR very much - I agree it may not be obvious at all. We purposely leave some schemas open for just this reason!
